app/src/main/kotlin/edu/stanford/bdh/engagehf/medication/data/MedicationRecommendationMapper.kt

```diff
@@ -11,18 +11,23 @@ class MedicationRecommendationMapper @Inject constructor(
 ) {
 
     fun map(documentSnapshot: DocumentSnapshot): MedicationRecommendation? {
-        val displayInformation = documentSnapshot.data?.get("displayInformation") as? JsonMap ?: return null
+        val displayInformation =
+            documentSnapshot.data?.get("displayInformation") as? JsonMap ?: return null
         val title = localizedMapReader.get("title", displayInformation)
         val description = localizedMapReader.get("description", displayInformation)
         val subtitle = localizedMapReader.get("subtitle", displayInformation)
-        val type = MedicationRecommendationType.from(localizedMapReader.get("type", displayInformation))
-        return if (title != null && description != null && subtitle != null) {
+        val videoPath = localizedMapReader.get("videoPath", displayInformation)
+        val type =
+            MedicationRecommendationType.from(localizedMapReader.get("type", displayInformation))
+        @Suppress("ComplexCondition")
+        return if (title != null && description != null && subtitle != null && videoPath != null) {
             MedicationRecommendation(
                 id = documentSnapshot.id,
                 title = title,
                 description = description,
                 subtitle = subtitle,
                 type = type,
+                videoPath = videoPath,
```

Comment:
I think we can allow nullable field here instead of ignoring the whole recommendation, and in case not available we just hide the info icon in ui instead.